src/lucky_record/model.cr

```diff
@@ -103,4 +107,89 @@ class LuckyRecord::Model
   macro association(table_name, foreign_key = nil)
     {% ASSOCIATIONS << {name: table_name.id, foreign_key: foreign_key} %}
   end
+
+  macro setup_correct_field_mappings_method
+    def ensure_correct_field_mappings!
+      ensure_existing_table!
+
+      fields = [
+        {% for field in FIELDS %}
+          { name: :{{field[:name]}}, nilable: {{ field[:nilable] }}, type: {{ field[:type] }} },
+        {% end %}
+      ]
+      ensure_fields_match_columns! fields
+    end
+  end
+
+  def ensure_existing_table!
+    table_names = LuckyRecord::Repo.tables_with_schema(excluding: "migrations")
+    return if table_names.includes?(@@table_name.to_s)
```

Comment:
Early returns can often cause misleading conditionals. Instead what if you extracted a class to handle this and use a regular if/else?

```crystal
# in the LuckyRecord::SchemaEnforcer module
class EnsureExistingTable
  def initialize(@table_name : Symbol)
    @table_names = LuckyRecord::Repo.tables_with_schema(excluding: "migrations")
  end

  def ensure_exists!
    if table_missing?
      # raise the error message
    end
  end

  private def table_missing?
    !@table_names.includes?(@table_name)
  end
end
```

I hope that makes sense :) LMK if I wasn't clear and I can try to clarify more